descope/auth.py

```diff
@@ -38,6 +38,7 @@ def __init__(
         project_id: str = None,
         public_key: str = None,
         skip_verify: bool = False,
+        management_key: str = None,
```

Comment:
it make sense to put `management_key` here 👍 

but I think for that long run, we should create a config class similar to go-sdk (for comparison - in go-sdk the `management` and `auth` are two different namespaces

the main motivation is separation of concerns (plus the similarity to `go-sdk` is also a minor advantage)

@guyp-descope wdyt?